control/plugin/client/grpc.go

```diff
@@ -241,12 +281,142 @@ func (g *grpcClient) CollectMetrics(mts []core.Metric) ([]core.Metric, error) {
 	return metrics, nil
 }
 
+func (g *grpcClient) UpdateCollectedMetrics(mts []core.Metric) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			Metrics_Arg: &rpc.MetricsArg{Metrics: NewMetrics(mts)},
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (g *grpcClient) UpdatePluginConfig(bytes []byte) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			Other: bytes,
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (g *grpcClient) UpdateCollectDuration(maxCollectDuration time.Duration) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			MaxCollectDuration: maxCollectDuration.Nanoseconds(),
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+func (g *grpcClient) UpdateMetricsBuffer(maxMetricsBuffer int64) error {
+	if g.stream != nil {
+		arg := &rpc.CollectArg{
+			MaxMetricsBuffer: maxMetricsBuffer,
+		}
+		err := g.stream.Send(arg)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *grpcClient) StreamMetrics(mts []core.Metric) (chan []core.Metric, chan error, error) {
+	arg := &rpc.CollectArg{
+		Metrics_Arg: &rpc.MetricsArg{Metrics: NewMetrics(mts)},
+	}
+	if len(mts) == 0 {
+		return nil, nil, errors.New("No metrics requested to stream")
+	}
+
+	mt := mts[0]
+	if cfg := mt.Config(); cfg != nil {
+		values := cfg.Table()
+		if values != nil {
+			maxCollectDuration, ok := values["MaxCollectDuration"]
+			if ok {
+				t, ok := maxCollectDuration.(*ctypes.ConfigValueInt)
+				if ok {
+					arg.MaxCollectDuration = int64(t.Value)
+				}
+			}
+			maxMetricsBuffer, ok := values["MaxMetricsBuffer"]
+			if ok {
+				t, ok := maxMetricsBuffer.(*ctypes.ConfigValueInt)
+				if ok {
+					arg.MaxMetricsBuffer = int64(t.Value)
+				}
+			}
+		}
+	}
+
+	s, err := g.streamCollector.StreamMetrics(context.Background())
+	if err != nil {
+		return nil, nil, err
+	}
+	err = s.Send(arg)
```

Comment:
Every call to stream metrics does require the arg to be sent. I'm not sure what you mean by 'send-event'. When establishing the stream connection we do need to send the args.